packages/apps/project/src/components/pages/ReportingView/ReportingView.tsx

```diff
@@ -48,26 +57,26 @@ const ReportingView = () => {
         fixedItems={
           <>
             <ProjectHeader />
-            {project?.isBusinessProject && (
+            {project?.isBusinessProject && showReporting && showActivities && (
               <ToggleSwitch
                 leftTitle={I18n.t('Project_Reporting')}
                 rightTitle={I18n.t('Project_Activities')}
                 onSwitch={() =>
                   setMode(_mode => {
-                    return _mode === modes.activities
-                      ? modes.reporting
-                      : modes.activities;
+                    return _mode === ReportingType.activities
+                      ? ReportingType.indicator
+                      : ReportingType.activities;
                   })
                 }
               />
             )}
           </>
         }
       />
-      {mode === modes.activities ? (
+      {mode === ReportingType.activities && showActivities ? (
         <ActivityListView />
       ) : (
-        <ReportingDetailsView />
```

Comment:
Only check current mode as you create a useEffect to switch mode depending on configuration